src/plugins/vis_default_editor/public/components/sidebar/sidebar.tsx

```diff
@@ -104,12 +110,12 @@ function DefaultEditorSideBar({
         aggs: state.data.aggs ? (state.data.aggs.aggs.map((agg) => agg.toJSON()) as any) : [],
       },
     });
-    eventEmitter.emit('updateVis');
+    embeddableHandler.reload();
```

Comment:
Instead of emitting an event from a particular editor, and handle it in visualize as:

![image](https://user-images.githubusercontent.com/31325372/84497347-441d8780-acb7-11ea-8444-cb1c680f8ab8.png)

reload it inside an editor outright (IMHO it's more evident)